### PR TITLE
Add the stree command

### PR DESCRIPTION
**Pull request by ekinanp**
Contributions to this project require sign-off consistent with the [Developers Certificate of Origin](https://developercertificate.org). This can be as simple as using `git commit -s` on each commit.

### CODE REVIEW

**Comment by ekinanp** [comment]
Example output:
```
➜  mnt616562680 stree
.
├── aws
│   └── [profile]
│       └── resources
│           ├── s3
│           │   └── [bucket]
│           │       ├── [prefix]
│           │       │   ├── [prefix]
│           │       │   └── [object]
│           │       └── [object]
│           └── ec2
│               └── instances
│                   └── [instance]
│                       ├── [console.out]
│                       ├── metadata.json
│                       └── fs
│                           ├── [dir]
│                           │   ├── [dir]
│                           │   └── [file]
│                           └── [file]
├── docker
│   ├── containers
│   │   └── [container]
│   │       ├── log
│   │       ├── metadata.json
│   │       └── fs
│   │           ├── [dir]
│   │           │   ├── [dir]
│   │           │   └── [file]
│   │           └── [file]
│   └── volumes
│       └── [volume]
│           ├── [dir]
│           │   ├── [dir]
│           │   └── [file]
│           └── [file]
└── kubernetes
    └── [context]
        └── [namespace]
            ├── pods
            │   └── [pod]
            └── persistentvolumeclaims
                └── [persistentvolumeclaim]
                    ├── [dir]
                    │   ├── [dir]
                    │   └── [file]
                    └── [file]
```

With multiple files:
```
➜  mnt633914967 stree aws docker
aws
└── [profile]
    └── resources
        ├── s3
        │   └── [bucket]
        │       ├── [prefix]
        │       │   ├── [prefix]
        │       │   └── [object]
        │       └── [object]
        └── ec2
            └── instances
                └── [instance]
                    ├── [console.out]
                    ├── metadata.json
                    └── fs
                        ├── [dir]
                        │   ├── [dir]
                        │   └── [file]
                        └── [file]
docker
├── containers
│   └── [container]
│       ├── log
│       ├── metadata.json
│       └── fs
│           ├── [dir]
│           │   ├── [dir]
│           │   └── [file]
│           └── [file]
└── volumes
    └── [volume]
        ├── [dir]
        │   ├── [dir]
        │   └── [file]
        └── [file]
```

File: cmd/stree.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	apitypes "github.com/puppetlabs/wash/api/types"
+	cmdutil "github.com/puppetlabs/wash/cmd/util"
+	"github.com/spf13/cobra"
+)
+
+func streeCommand() *cobra.Command {
+	streeCmd := &cobra.Command{
+		Use:   "stree [<path>...]",
+		Short: "Displays the entry's stree (schema-tree)",
+		Long: `Displays the entry's stree (schema-tree), which is a high-level overview of the entry's
```
Comment (ekinanp) [inline cmd/stree.go]: Think this could be more complete, but not sure how to word this better.

File: cmd/stree.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		fill(stree, schema)
+		stree.SetValue(path)
+		cmdutil.Print(stree.String())
+	}
+	return exitCode{0}
+}
+
+func fill(stree treeprint.Tree, schema apitypes.EntrySchema) treeprint.Tree {
+	value := schema.Label
+	if len(value) == 0 {
```
Comment (ekinanp) [inline cmd/stree.go]: Think we should just require `Label` (i.e. panic if it is not provided). I see no good reason to default to printing Type.

File: api/types/schema.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package apitypes
+
+import "github.com/puppetlabs/wash/plugin"
+
+// EntrySchema describes an entry's schema, which
+// is what's returned by the /fs/schema endpoint.
+type EntrySchema = plugin.EntrySchema
```
Comment (ekinanp) [inline api/types/schema.go]: I kind of want to adopt this pattern for `plugin.EntryAttributes` + the actions. It'd be nice if the cmd's did not depend on the `plugin` package, but I can see the redundancy behind that (esp. w.r.t the actions) so am happy to remove this if that's preferred.

File: cmd/stree.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+		value = schema.Type
+	}
+	if !schema.Singleton {
+		value = fmt.Sprintf("[%v]", value)
```
Comment (ekinanp) [inline cmd/stree.go]: We could add some more info here (e.g. the entry's supported actions), but not sure if that'd be useful or if we should just make that an option.

File: api/types/schema.go (hunk 3 above)
Comment (MikaelSmith) [reply api/types/schema.go]: I'm ok with that pattern. I think I've used it sparingly elsewhere.

Comment (ekinanp) [reply api/types/schema.go]: Ok. I'll put up a small cleanup PR after this to adopt this pattern where it makes sense.

File: cmd/stree.go (hunk 4 above)
Comment (MikaelSmith) [reply cmd/stree.go]: I'd definitely be interested in entry's supported actions. Making it an additional option feels consistent with other tools in the ecosystem.

Comment (ekinanp) [reply cmd/stree.go]: Ok, I'll file an issue.

**Comment by MikaelSmith** [comment]
Supports #291.

**Pull request merged**